pandas/tests/test_categorical.py

```diff
@@ -3822,6 +3822,34 @@ def test_cat_equality(self):
         pytest.raises(TypeError, lambda: a > b)
         pytest.raises(TypeError, lambda: b > a)
 
+    def test_unordered_different_order_equal(self):
+        # https://github.com/pandas-dev/pandas/issues/16014
+        c1 = Categorical(['a', 'b'], categories=['a', 'b'], ordered=False)
+        c2 = Categorical(['a', 'b'], categories=['b', 'a'], ordered=False)
```

Comment:
Hopefully the new version isn't too weird: https://github.com/pandas-dev/pandas/pull/16339/files#diff-ed4f442894a2f521dfac3193a3a8d8a0R3825